cpp/oneapi/dal/algo/rbf_kernel/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -15,60 +15,110 @@
-                                 const descriptor_t& desc,
-                                 const table& x,
-                                 const table& y) {
+static result_t compute(const context_gpu& ctx, const descriptor_t& desc, const input_t& input) {
+    const auto x = input.get_x();
+    const auto y = input.get_y();
+
     auto& queue = ctx.get_queue();
-    interop::execution_context_guard guard(queue);
 
-    const int64_t row_count_x = x.get_row_count();
-    const int64_t row_count_y = y.get_row_count();
+    const std::int64_t x_row_count = x.get_row_count();
+    const std::int64_t y_row_count = y.get_row_count();
 
-    dal::detail::check_mul_overflow(row_count_x, row_count_y);
-    auto arr_values =
-        array<Float>::empty(queue, row_count_x * row_count_y, sycl::usm::alloc::device);
+    ONEDAL_ASSERT(x.get_column_count() == y.get_column_count());
+    dal::detail::check_mul_overflow(x_row_count, y_row_count);
 
-    const auto daal_x = interop::convert_to_daal_table(queue, x);
-    const auto daal_y = interop::convert_to_daal_table(queue, y);
-    const auto daal_values =
-        interop::convert_to_daal_table(queue, arr_values, row_count_x, row_count_y);
+    const double sigma = desc.get_sigma();
+    const Float coeff = static_cast<Float>(-0.5 / (sigma * sigma));
 
-    daal_rbf_kernel::Parameter daal_parameter(desc.get_sigma());
-    daal_rbf_kernel_t<Float>().compute(daal_x.get(),
-                                       daal_y.get(),
-                                       daal_values.get(),
-                                       &daal_parameter);
+    const auto ndarray_x =
+        pr::flatten_table<Float, row_accessor>(queue, x, sycl::usm::alloc::device);
 
-    return result_t().set_values(
-        dal::detail::homogen_table_builder{}.reset(arr_values, row_count_x, row_count_y).build());
-}
+    const auto ndarray_y =
+        pr::flatten_table<Float, row_accessor>(queue, y, sycl::usm::alloc::device);
 
-template <typename Float>
-static result_t compute(const context_gpu& ctx, const descriptor_t& desc, const input_t& input) {
-    return call_daal_kernel<Float>(ctx, desc, input.get_x(), input.get_y());
+    auto ndarray_res =
+        pr::ndarray<Float, 2>::empty(queue, { x_row_count, y_row_count }, sycl::usm::alloc::device);
+
+    auto ndarray_sqr_x =
+        pr::ndarray<Float, 1>::empty(queue, { x_row_count }, sycl::usm::alloc::device);
+    auto ndarray_sqr_y =
+        pr::ndarray<Float, 1>::empty(queue, { y_row_count }, sycl::usm::alloc::device);
+
+    auto reduce_x_event =
+        reduce_by_rows(queue, ndarray_x, ndarray_sqr_x, pr::sum<Float>{}, pr::square<Float>{});
+    auto reduce_y_event =
+        reduce_by_rows(queue, ndarray_y, ndarray_sqr_y, pr::sum<Float>{}, pr::square<Float>{});
+
+    constexpr Float alpha = -2.0;
+    constexpr Float beta = 0.0;
+    auto gemm_event = gemm(queue, ndarray_x, ndarray_y.t(), ndarray_res, alpha, beta);
```

Comment:
This can be moved inside `compute_rbf` function since logically this is a part of rbf computation